barcode_utils.py

```diff
@@ -0,0 +1,166 @@
+from barcode import *
+
+import base64
+import struct
+import skip32
+import code128
+import os
+
+
+# TODO: currently broken, need to rewrite in the context of uber
+def generate_barcode_csv(filename):
+    badge_types = self.badge_types['badge_types']
+
+    # TODO: modify so we read the config directly from uber.
+    lines = []
+    for badge_type, ranges in badge_types.items():
+        range_start = int(ranges['range_start'])
+        range_end = int(ranges['range_end'])
+        lines = lines + self.generate_barcode_nums(range_start, range_end)
+
+    f = open(filename,'w')
+
+    for line in lines:
+        f.write(line + os.linesep)
+
+    f.close()
+
+
+# TODO: currently broken, need to rewrite in the context of uber
+def generate_barcode_nums(range_start, range_end):
+    generated_lines = []
+    seen_barcodes = []
+    for badge_num in range(range_start, range_end+1):
+        barcode_num = generate_barcode_from_badge_num(
+            badge_num=int(badge_num),
+            event_id=self.event_id,
+            salt=self.salt,
+            key=self.secret_key
+        )
+
+        line = "{badge_num},{barcode_num}".format(
+            badge_num=badge_num,
+            barcode_num=barcode_num,
+        )
+        generated_lines.append(line)
+
+        # ensure that we haven't seen this value before
+        if barcode_num in seen_barcodes:
+            raise ValueError('COLLISION: generated a badge# that\'s already been seen')
+        else:
+            seen_barcodes.append(barcode_num)
+
+    return generated_lines
+
```

Comment:
hm, yea that could work, I'll check into it.  One question we should address is whether global c should pickup config values from plugins automatically.  I think that's probably a good idea? so that a page in core could enumerate values injected by plugins.